src/as_execution/mod.rs

```diff
@@ -60,36 +59,32 @@ impl ASModule {
 }
 
 pub(crate) fn init_engine(limit: u64, gas_costs: GasCosts) -> Engine {
-    // We use the Singlepass compiler because it is fast and adapted to blockchains
-    // See https://docs.rs/wasmer-compiler-singlepass/latest/wasmer_compiler_singlepass/
-    let mut compiler_config = Singlepass::new();
+    // Use Cranelift (in opposition to Singlepass)
+    // Because of module caching we can run longer compilations to have better optimizations
+    // Executions are happening way more often than compilations hence that choice
+    let mut compiler_config = Cranelift::new();
```

Comment:
I would keep the option to init an engine with != compiler

Reply:
Concerning the compiler change, it has been decided in https://github.com/massalabs/massa/discussions/3560#discussioncomment-4960194 that we will keep Singlepass for now, will be reverting those changes

Reply:
Done